packages/smart-forms-renderer/src/stories/itemTypes/Choice.stories.tsx

```diff
@@ -39,48 +37,166 @@ export default meta;
+    expect(result[0].valueCoding).toEqual(expect.objectContaining(targetCoding));
   }
 };
 
 export const ChoiceAnswerOptionBasicResponse: Story = {
   args: {
     questionnaire: qChoiceAnswerOptionBasic,
     questionnaireResponse: qrChoiceAnswerOptionBasicResponse
+  }, play: async ({ canvasElement }) => {
+    const inputText = await getInputText(canvasElement, targetlinkId);
+
+    expect(inputText).toBe(targetCoding.display)
   }
 };
 
+const valueSetTargetId = 'gender'
+
+const valueSetTargetCoding = {
+  code: "female",
+  display: "Female",
+  system: "http://hl7.org/fhir/administrative-gender",
+}
+
+const qValueSetBasic = questionnaireFactory([
+  {
+    linkId: valueSetTargetId,
+    text: 'Gender',
+    type: 'choice',
+    repeats: false,
+    answerValueSet: 'http://hl7.org/fhir/ValueSet/administrative-gender'
+  }
+])
+
 export const ChoiceAnswerValueSetBasic: Story = {
   args: {
-    questionnaire: qChoiceAnswerValueSetBasic
+    questionnaire: qValueSetBasic
+  },
+  play: async ({ canvasElement }) => {
+    await chooseSelectOption(
+      canvasElement,
+      valueSetTargetId,
+      valueSetTargetCoding.display
+    );
+    const result = await getAnswers(valueSetTargetId);
+    expect(result).toHaveLength(1);
+    expect(result[0].valueCoding).toEqual(
+      expect.objectContaining(valueSetTargetCoding)
+    );
   }
 };
 
+
 export const ChoiceAnswerValueSetBasicResponse: Story = {
   args: {
-    questionnaire: qChoiceAnswerValueSetBasic,
-    questionnaireResponse: qrChoiceAnswerValueSetBasicResponse
+    questionnaire: qValueSetBasic,
+    questionnaireResponse: qrFactory([
+      {
+        linkId: 'gender',
+        text: 'Gender',
+        answer: [{ valueCoding: valueSetTargetCoding }]
+      }
+    ])
+  },
+  play: async ({ canvasElement }) => {
+    const inputText = await getInputText(canvasElement, valueSetTargetId);
+    expect(inputText).toBe(valueSetTargetCoding.display);
+  }
+};
+
+// Story for ChoiceSelectAnswerOptions Using InitialSelected field set
+export const ChoiceAnswerOptionsUsingInitialSelected: Story = {
+  args: {
+    questionnaire: questionnaireFactory([
+      {
+        text: 'Type',
+        type: 'choice',
+        linkId: 'awsHallucinationType',
+        repeats: false,
+        answerOption: [
+          {
+            valueCoding: {
+              code: 'N',
+              system: 'http://fhir.medirecords.com/CodeSystem/awsHallucinationType',
+              display: 'None'
+            }
+          },
+          {
+            valueCoding: {
+              code: 'T',
+              system: 'http://fhir.medirecords.com/CodeSystem/awsHallucinationType',
+              display: 'Test-Selected'
+            },
+            initialSelected: true
+          }
+        ]
+      }
+    ]),
+  }, play: async ({ canvasElement }) => {
+    const inputText = await getInputText(canvasElement, "awsHallucinationType");
+    expect(inputText).toBe("Test-Selected")
```

Comment:
@Vlpros also please check that qr has answers 
```
{
              code: 'T',
              system: 'http://fhir.medirecords.com/CodeSystem/awsHallucinationType',
              display: 'Test-Selected'
            },
```
after initial selection. Please, re-use pattern that you used in the previous tests, when codings are stored as separate variables